llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1600,7 +1598,13 @@ bool MemCpyOptPass::performStackMoveOptzn(Instruction *Load, Instruction *Store,
   if (!CaptureTrackingWithModRef(SrcAlloca, SrcModRefCallback))
     return false;
 
-  // We can do the transformation. First, align the allocas appropriately.
+  // We can do the transformation. First, move the SrcAlloca to the entry point
+  // if it's not dominator for all uses. After that SrcAlloca becomes the
+  // dominator, because it's in the entry BB.
```

Comment:
This comment makes it sound like we're moving the alloca to the entry BB. The alloca is already in the entry BB, we're just moving it to the start of it.